### PR TITLE
Added support for command line options

### PR DESCRIPTION
**Pull request by bzar**
1. Refactored `index.js` into `bootstrap` and `main` (before and after having a planmill client)
1. Added `lib/options.js` for handling command line options with `command-line-args` and `command-line-usage`
1. Refactored `lib/planmill.js` to be instantiated with a configuration
    1. Nicer than passing opts to all functions?
1. Implemented configuration file setting using `path.resolve` with `./config.json` as default
    1. Should have better autodetection from standard locations if not explicitly set

### CODE REVIEW

**Comment by bzar** [comment]
Also added a system for adding config file default search locations. Paths should be added for various platforms' variants

File: index.js [hunk 1]
```diff
@@ -1,70 +1,93 @@
...
+  } catch(e) {
+    console.log('ERROR:', e.message)
+    console.log(options.usage())
+    return null
+  }
+}
 
-// Start the spinner
-loader.start()
+const main = (pm) => {
```
Comment (leomelin) [inline index.js]: unnecessary parentheses

File: lib/options.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    header: 'planmill-cli',
+    content: 'A tool for interacting with [italic]{planmill}'
+  },
+  {
+    header: 'Options',
+    optionList: optionDefinitions
+  }
+]
+
+const parse = (argv) => commandLineArgs(optionDefinitions, argv)
```
Comment (leomelin) [inline lib/options.js]: unnecessary parentheses

File: lib/planmill.js [hunk 3]
```diff
@@ -1,129 +1,163 @@
...
 
-let auth = null
-const getInit = () => {
-  return !auth ? saveAuthentication() : Promise.resolve()
+const configSearchLocations = () => [
+  process.env.HOME + '/.config/planmill/config.json',
+  './config.json',
+]
+
+const getConfig = (configPath) => {
```
Comment (leomelin) [inline lib/planmill.js]: unnecessary parentheses

File: lib/planmill.js [hunk 4]
```diff
@@ -1,129 +1,163 @@
+const path = require('path');
 const rp = require('request-promise-native')
-const config = require('../config.json')
 const utils = require('./utils')
 
-let auth = null
-const getInit = () => {
-  return !auth ? saveAuthentication() : Promise.resolve()
+const configSearchLocations = () => [
+  process.env.HOME + '/.config/planmill/config.json',
```
Comment (leomelin) [inline lib/planmill.js]: Nice 👍

File: lib/planmill.js [hunk 5]
```diff
@@ -1,129 +1,163 @@
...
+  './config.json',
+]
+
+const getConfig = (configPath) => {
+  if(configPath) {
```
Comment (leomelin) [inline lib/planmill.js]: Missing space. Also other linter errors - run `yarn run lint`

File: lib/planmill.js [hunk 6]
```diff
@@ -1,129 +1,163 @@
...
-        'cache-control': 'no-cache',
-        authorization: `Bearer ${auth.access_token}`
-      },
-      qs: query,
-      json: true
-    }).then(body => Promise.resolve(body))
-  },
-  post: (path, body, query) => {
-    return rp.post(`${config.apiEndpoint}/${path}`, {
+const instance = (options) => {
```
Comment (leomelin) [inline lib/planmill.js]: unnecessary parentheses

**Review by leomelin — CHANGES_REQUESTED** [review]
👍 Lots of linter errors, fix pls

**Pull request merged**